### PR TITLE
[INLONG-10601][Manager] Optimize the agent task configuration process

### LINKED ISSUE #10601: [Improve][Manager] Optimize the agent task configuration process
### Description

Optimize the agent task configuration process.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
Fixes #10601

### Motivation

Optimize the agent task configuration process

### Modifications

1.Add a version number to the agent configuration pull interface to control whether the task needs to be changed, in order to prevent inconsistent configuration pulled by the agent from the manager when the manager is in multiple clusters.

2.Optimize the agent configuration pull process, unify task configurations into one table, and the agent pull interface only operates on this table to prevent low pull efficiency caused by multi table queries.

3.Agent cache is scheduled to pull in batches from the database to prevent cache breakdown.

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 1]
```diff
@@ -266,6 +269,12 @@ private void startHeartbeatTask() {
         }
     }
 
+    private void setReloadTimer() {
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        long reloadInterval = 60000L;
+        executorService.scheduleAtFixedRate(this::reload, reloadInterval, reloadInterval, TimeUnit.MILLISECONDS);
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:275]: scheduleAtFixedDelay

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 2]
```diff
@@ -293,6 +302,43 @@ public void report(TaskRequest request) {
         }
     }
 
+    @Transactional(rollbackFor = Exception.class)
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:305]: No need

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 3]
```diff
@@ -293,6 +302,43 @@ public void report(TaskRequest request) {
...
+    public void reload() {
+        LOGGER.debug("start to reload agent task config.");
+        try {
+            Map<String, TaskResult> tempTaskConfigMap = new ConcurrentHashMap<>();
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:309]: newTaskConfigMap

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 4]
```diff
@@ -266,6 +269,12 @@ private void startHeartbeatTask() {
         }
     }
 
+    private void setReloadTimer() {
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        long reloadInterval = 60000L;
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:274]: redundant variable

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 5]
```diff
@@ -293,6 +302,43 @@ public void report(TaskRequest request) {
...
+                        tempTaskConfigMap.putIfAbsent(key, taskResult);
+                    }
+                    AgentConfigInfo agentConfigInfo = JsonUtils.parseObject(agentTaskConfigEntity.getConfigParams(),
+                            AgentConfigInfo.class);
+                    if (agentConfigInfo != null) {
+                        agentConfigInfo.setVersion(agentTaskConfigEntity.getVersion());
+                        tempAgentConfigMap.putIfAbsent(key, agentConfigInfo);
+                    }
+                } catch (Exception e) {
+                    LOGGER.error("failed to get agent task confgi for agent ip={}, cluster name={}",
```
Comment (justinwwhuang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:329]: confgi -> config

Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:329]: confgi->config

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java [hunk 6]
```diff
@@ -346,4 +403,222 @@ public void syncSourceFieldInfo(SourceRequest request, String operator) {
...
+
+    private void updateAgentTaskConfig(SourceRequest request, String operator) {
+        try {
+            if (SourceType.AUTO_PUSH.equals(request.getSourceType())) {
+                return;
+            }
+            final String clusterName = request.getInlongClusterName();
+            final String ip = request.getAgentIp();
+            final String uuid = request.getUuid();
+            if (StringUtils.isBlank(clusterName) || StringUtils.isBlank(ip)) {
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:415]: It's better to print a log containing clusterName and ip.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java [hunk 7]
```diff
@@ -346,4 +403,222 @@ public void syncSourceFieldInfo(SourceRequest request, String operator) {
...
+            String md5 = DigestUtils.md5Hex(GSON.toJson(taskResult));
+            taskResult.setMd5(md5);
+            taskResult.setCode(AgentResponseCode.SUCCESS);
+            agentTaskConfigEntity.setAgentIp(request.getAgentIp());
+            agentTaskConfigEntity.setClusterName(request.getInlongClusterName());
+            agentTaskConfigEntity.setTaskParams(objectMapper.writeValueAsString(taskResult));
+
+            LOGGER.debug("begin to get agent config info for {}", request);
+            Set<String> tagSet = new HashSet<>(16);
+            List<StreamSourceEntity> sourceEntities =
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:465]: not used.

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:465]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java (hunk 6 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:415]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 5 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:329]: 60000L

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:329]: 60000L

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:329]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 4 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:274]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 3 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:309]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 2 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:305]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:275]: Fixed.

**Review by justinwwhuang — APPROVED** [review]
LGTM

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/TemplateServiceImpl.java [hunk 8]
```diff
@@ -116,7 +116,7 @@ public TemplateInfo get(String templateName, String operator) {
         TemplateEntity templateEntity = templateEntityMapper.selectByName(templateName);
         if (templateEntity == null) {
             LOGGER.error("inlong template not found by template name={}", templateName);
-            throw new BusinessException(ErrorCodeEnum.TEMPLATE_INFO_INCORRECT);
+            throw new BusinessException(ErrorCodeEnum.TENANT_NOT_EXIST);
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/TemplateServiceImpl.java:119]: TEMPLATE_NOT_EXIST?

**Review by aloyszhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/TemplateServiceImpl.java (hunk 8 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/stream/TemplateServiceImpl.java]: Fixed.

**Review by justinwwhuang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**